commit 51bb73c6 causes disconnects with AMF

## Description
After testing for a while, I found that there are disconnect problems with https://github.com/alvr-org/ALVR/commit/51bb73c65b75d43738dbf1848d9326010398802a while using AMF. Sometimes the server can't reconnect.

ALVR-log looks like this:

```
12:39:18.196051063      [ERROR] What happened: panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: null, expected f32", line: 1, column: 314)', alvr/server/src/logging_backend.rs:50:48 Backtrace: 0: alvr_common::logging::set_panic_hook::{{closure}} 1: as core::ops::function::Fn>::call at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/alloc/src/boxed.rs:2002:9 2: std::panicking::rust_panic_with_hook at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/std/src/panicking.rs:692:13 3: std::panicking::begin_panic_handler::{{closure}} at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/std/src/panicking.rs:579:13 4: std::sys_common::backtrace::__rust_end_short_backtrace at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/std/src/sys_common/backtrace.rs:137:18 5: rust_begin_unwind at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/std/src/panicking.rs:575:5 6: core::panicking::panic_fmt at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/core/src/panicking.rs:64:14 7: core::result::unwrap_failed at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/core/src/result.rs:1790:5 8: alvr_server::logging_backend::init_logging::{{closure}} 9: ::log 10: log::__private_api_log 11: alvr_events::send_event 12: as core::future::future::Future>::poll 13: alvr_sockets::util::spawn_cancelable::{{closure}}::{{closure}} 14: tokio::runtime::task::core::Core::poll 15: tokio::runtime::task::harness::Harness::poll 16: tokio::runtime::scheduler::multi_thread::worker::Context::run_task 17: tokio::runtime::scheduler::multi_thread::worker::Context::run 18: tokio::macros::scoped_tls::ScopedKey::set 19: tokio::runtime::scheduler::multi_thread::worker::run 20: tokio::loom::std::unsafe_cell::UnsafeCell::with_mut 21: tokio::runtime::task::core::Core::poll 22: tokio::runtime::task::harness::Harness::poll 23: tokio::runtime::blocking::pool::Inner::run 24: std::sys_common::backtrace::__rust_begin_short_backtrace 25: core::ops::function::FnOnce::call_once{{vtable.shim}} 26: as core::ops::function::FnOnce>::call_once at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/alloc/src/boxed.rs:1988:9 27: as core::ops::function::FnOnce>::call_once at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/alloc/src/boxed.rs:1988:9 28: std::sys::unix::thread::Thread::new::thread_start at /rustc/61a415be590113b4935464ef0aaf3b4e7713a077/library/std/src/sys/unix/thread.rs:108:17 29: 30:
12:39:18.196135133      [WARN]  At alvr/sockets/src/lib.rs:41: task 93 panicked
12:39:19.500956575      [INFO]  SessionUpdated
12:39:19.501000465      [INFO]  Session
12:39:19.888629192      [INFO]  Initial UDP buffer size: send: 212992B, recv: 212992B
12:39:19.888671102      [INFO]  New UDP buffer send size: 425984B
12:39:19.888674572      [INFO]  New UDP buffer recv size: 425984B
12:39:19.888683862      [INFO]  ClientConnected
12:39:19.888837872      [INFO]  Serial Number: 1WMGH000XX0000
12:39:19.888842802      [INFO]  Model Number: Miramar
12:39:19.888845152      [INFO]  Render Target: 4288 2400
12:39:19.888848172      [INFO]  Seconds from Vsync to Photons: 0.005000
12:39:19.888850282      [INFO]  Refresh Rate: 60
12:39:19.994834346      [INFO]  Client 1824.client.alvr: [Warning] Video frame queue underflow!
12:39:20.056991212      [INFO]  Client 1824.client.alvr: [Info] Last log line repeated 30 times
12:39:20.057012282      [INFO]  Client 1824.client.alvr: [Info] Decoder dequeue event: OutputBuffersChanged
12:39:20.057832572      [INFO]  Client 1824.client.alvr: [Warning] Video frame queue underflow!
12:39:20.072962031      [INFO]  Client 1824.client.alvr: [Info] Last log line repeated 8 times
12:39:20.073127801      [INFO]  Client 1824.client.alvr: [Info] Decoder dequeue event: OutputFormatChanged
12:39:20.073564491      [INFO]  AMF: 0123-00-13 12:39:20 FBFFF6C0 [EncodeQueueVkImpl] Warning: FsEncodeQueueVulkan::Flush re-CreateQueue happened!
12:39:20.073590921      [INFO]  AMF: 0123-00-13 12:39:20 FBFFF6C0 [AMFEncoderCoreBaseImpl] Warning: ReleaseThroughput() - Failed to release throughput.
12:39:20.076734291      [ERROR] error in encoder thread: Vulkan: VK_NOT_READYat cpp/platform/linux/EncodePipelineAMF.cpp:468
```

--- a/alvr/server/cpp/platform/linux/EncodePipelineAMF.cpp
+++ b/alvr/server/cpp/platform/linux/EncodePipelineAMF.cpp
@@ -422,19 +422,32 @@ void EncodePipelineAMF::PushFrame(uint64_t targetTimestampNs, bool idr)
 
     vkEndCommandBuffer(m_commandBuffer);
 
-    VkSemaphore waitSemaphore = m_render->GetOutput().semaphore;
-    VkPipelineStageFlags waitStage = VK_PIPELINE_STAGE_TRANSFER_BIT;
+    uint32_t waitCount = 0;
+    VkSemaphore waitSemaphores[2];
+    VkPipelineStageFlags waitStages[2];
+
+    waitSemaphores[waitCount] = m_render->GetOutput().semaphore;
+    waitStages[waitCount] = VK_PIPELINE_STAGE_TRANSFER_BIT;
+    waitCount++;
+
+    if (surfaceVk->Sync.bSubmitted) {
+        waitSemaphores[waitCount] = surfaceVk->Sync.hSemaphore;
+        waitStages[waitCount] = VK_PIPELINE_STAGE_TRANSFER_BIT;
+        waitCount++;
+    }
 
     VkSubmitInfo submitInfo = {};
     submitInfo.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
-    submitInfo.waitSemaphoreCount = 1;
-    submitInfo.pWaitSemaphores = &waitSemaphore;
-    submitInfo.pWaitDstStageMask = &waitStage;
+    submitInfo.waitSemaphoreCount = waitCount;
+    submitInfo.pWaitSemaphores = waitSemaphores;
+    submitInfo.pWaitDstStageMask = waitStages;
     submitInfo.signalSemaphoreCount = 1;
     submitInfo.pSignalSemaphores = &surfaceVk->Sync.hSemaphore;
     submitInfo.commandBufferCount = 1;
     submitInfo.pCommandBuffers = &m_commandBuffer;
     VK_CHECK(vkQueueSubmit(m_render->m_queue, 1, &submitInfo, nullptr));
+
+    surfaceVk->Sync.bSubmitted = true;
 
     m_targetTimestampNs = targetTimestampNs;
 

